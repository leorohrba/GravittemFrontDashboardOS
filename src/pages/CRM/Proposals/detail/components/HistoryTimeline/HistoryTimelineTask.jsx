import { formatTaskDuration } from '@utils'
import { Card, Col, Row, Timeline } from 'antd'
import PropTypes from 'prop-types'
import React, { useEffect, useState } from 'react'
import HistoryTimelineDate from './HistoryTimelineDate'

export default function HistoryTimelineTask({
  date,
  title,
  action,
  oldValue,
  newValue,
  userName,
}) {
  const [task, setTask] = useState(null)
  const [comments, setComments] = useState([])

  useEffect(() => {
    try {
      const taskWork = JSON.parse(newValue)
      setTask(taskWork)
      if (taskWork.observation) {
        setComments(taskWork.observation.split('\n'))
      }
      // eslint-disable-next-line no-console
    } catch (error) {
      // console.log(error)
    }
  }, [newValue])

  return (
    <React.Fragment>
      {task && (
        <Timeline.Item color="gray">
          <HistoryTimelineDate date={date} />
          <Card className="mt-3" bodyStyle={{ padding: 0 }}>
            <h3
              className="text-white p-4"
              style={{
                backgroundColor:
                  action === 'Include'
                    ? '#1976d2'
                    : action === 'Exclude'
                    ? '#c53030'
                    : '#4CAF50',
              }}
            >
              <i
                className={`fa fa-${
                  action === 'Exclude' ? 'calendar-times-o' : 'calendar-check-o'
                } mr-3 ml-1 fa-lg`}
                aria-hidden="true"
              />
              <b>{title}</b>
              <span>{` - por ${userName}`}</span>
            </h3>
            <div className="p-4">
              {task.subject && (
                <h4>
                  <span className="font-bold">{task.subject}</span>
                </h4>
              )}
              {task.expectedDuration && (
                <h4>
                  <span className="font-bold">Duração: </span>
                  <span>{formatTaskDuration(task.expectedDuration)}</span>
                </h4>
              )}
              {task.observation && (
                <h4>
                  <Row type="flex">
                    <Col className="mr-2">
                      <span className="font-bold">Observações: </span>
                    </Col>
                    <Col>
                      {comments.map((comment, index) => (
                        <Row key={index}>
                          <Col>
                            {comment && (
                              <React.Fragment>{comment}</React.Fragment>
                            )}
                            {!comment && <br />}
                          </Col>
                        </Row>
                      ))}
                    </Col>
                  </Row>
                </h4>
              )}
            </div>
          </Card>
        </Timeline.Item>
      )}
    </React.Fragment>
  )
}

HistoryTimelineTask.propTypes = {
  action: PropTypes.string,
  date: PropTypes.object,
  newValue: PropTypes.string,
  oldValue: PropTypes.string,
  title: PropTypes.string,
  userName: PropTypes.string,
}
