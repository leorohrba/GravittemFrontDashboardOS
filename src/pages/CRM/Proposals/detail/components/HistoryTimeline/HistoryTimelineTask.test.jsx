import { render } from '@testing-library/react'
import HistoryTimelineTask from './HistoryTimelineTask'

it('with observation', async () => {
  const { getByText } = render(
    <HistoryTimelineTask
      date={{}}
      title="title"
      action="Send"
      oldValue="old"
      newValue='{ "observation":"test"}'
      userName="user"
    />,
  )
  expect(getByText('test')).toBeInTheDocument()
})
it('test Include', async () => {
  const { getByText } = render(
    <HistoryTimelineTask
      date={{}}
      title="title"
      action="Include"
      oldValue="old"
      newValue='{ "observation":"test"}'
      userName="user"
    />,
  )
  expect(getByText('test')).toBeInTheDocument()
})
it('test Exclude', async () => {
  const { getByText } = render(
    <HistoryTimelineTask
      date={{}}
      title="title"
      action="Exclude"
      oldValue="old"
      newValue='{ "observation":"test", "task": {"subject": "test", "expectedDuration": 2 }}'
      userName="user"
    />,
  )
  expect(getByText('test')).toBeInTheDocument()
})

it('without observation', async () => {
  render(
    <HistoryTimelineTask
      date={{}}
      title="title"
      action="Send"
      oldValue="old"
      newValue=""
      userName="user"
    />,
  )
  const node = (
    <body>
      <div />
    </body>
  )
  expect(node).toMatchSnapshot()
})
