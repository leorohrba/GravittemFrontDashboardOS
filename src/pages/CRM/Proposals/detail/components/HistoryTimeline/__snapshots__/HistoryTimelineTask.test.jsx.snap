// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`without observation 1`] = `
<body>
  <div />
</body>
`;
